chatterino2: init at unstable-2019-05-11

rexim opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#sec-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change
Chatterino is a chat client for Twitch chat. It aims to be an improved/extended version of the Twitch web chat. It's actively used in the Twitch community.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---
cc @fourtf @pajlada

artemist commented on pkgs/applications/networking/irc/chatterino2/default.nix:
```diff
@@ -0,0 +1,33 @@
+  version = "2.0.4";
+  name = "${pname}-${version}";
+  src = fetchgit {
+    url = "https://github.com/fourtf/${pname}.git";
```
You can use `fetchFromGitHub` instead of `fetchgit` here.

rexim replied:
Oh, ok! After a brief googling I came to a conclusion that `fetchFromGitHub` doesn't support submodules. Just double checked the source code and noticed `fetchSubmodules`. I'll try it now. Thanks!

artemist commented:
This package otherwise looks good. Thanks for changing that so quickly!

symphorien commented on pkgs/applications/networking/irc/chatterino2/default.nix:
```diff
@@ -0,0 +1,34 @@
+{ stdenv, pkgconfig, fetchFromGitHub, qt5, boost, openssl }:
```
please use qtbase etc. as arguments instead of qt5: see https://nixos.org/nixpkgs/manual/#ssec-qt-applications for details.

rexim replied:
@symphorien oh, that is interesting! Thank you for the link!

aanderse commented on maintainers/maintainer-list.nix:
```diff
@@ -4091,6 +4091,11 @@
     github = "retrry";
     name = "Tadas Barzdžius";
   };
+  rexim = {
```
Please include in a separate commit

aanderse commented on pkgs/applications/networking/irc/chatterino2/default.nix:
```diff
@@ -0,0 +1,34 @@
+stdenv.mkDerivation rec {
+  pname = "chatterino2";
+  version = "2.0.4";
+  name = "${pname}-${version}";
```
This is taken care of automatically so you can omit

aanderse commented on pkgs/applications/networking/irc/chatterino2/default.nix:
```diff
@@ -0,0 +1,34 @@
+  };
+  nativeBuildInputs = [ qmake pkgconfig ];
+  buildInputs = [ qtbase qtsvg qtmultimedia boost openssl ];
+  installPhase = ''
```
Does the `.pro` file included not take care of installing the binary automatically? If not I'd suggest that might be worth a PR upstream...

pajlada replied:
Fixed upstream in the relevant branch, `make install` now installs all relevant stuff to /usr/local (modifiable with `qmake PREFIX=/usr`)

aanderse commented on pkgs/applications/networking/irc/chatterino2/default.nix:
```diff
@@ -0,0 +1,34 @@
+  '';
+    homepage = "https://github.com/fourtf/chatterino2";
+    license = licenses.mit;
+    platforms = [ "x86_64-linux" ];
```
There was no mention of  64 bit only support in the README and reviewing the `.pro` makes me think you could use `platforms.unix` here instead.

aanderse reviewed (changes requested):
Thanks for choosing to contribute to nixpkgs! I have included a few suggestions which I hope will be helpful.

After you're ready with this PR you should rewrite the git history to be 2 commits, the first of which is you adding yourself to the maintainer list and the second adding the package (ie. everything else squashed).

Please let me know if you have any questions.

pajlada commented:
is "irc" the appropriate category for this package?
The technology that drives the client is IRC, yes, but it's a chat client that can right now only join twitch.tv chats

rexim commented:
@pajlada yeah, I agree. Maybe [instant-messengers](https://github.com/NixOS/nixpkgs/tree/master/pkgs/applications/networking/instant-messengers) is a better category?

rexim commented:
@aanderse thank you for the remarks! No questions so far. My NixOS machine recently broke, I'll address the remarks as soon as I get it back working. Within a week or so.

pajlada commented:
@rexim I'm not the most familiar with the categories of packages in NixOS, but from what I can see chatterino fits better under instant-messengers yeah

rexim committed:
Add rexim to maintainers

rexim committed:
chatterino2: init at unstable-2019-05-11

rexim commented:
@aanderse I hope I addressed most of the concerns. However there is one more thing. I recenly learned that the version in tag `2.0.4` is very old and the most relevant changes are in the `nightly` branch. So the application has kind of a rolling release nature. I changed the `version` to `2.0.4-nightly` and rev to the latest in `nightly` and will periodically bump it when enough changes in `nightly` are accumulated. Is that an appropriate way of maintaining packages in nixpkgs?

aanderse commented:
@rexim Awesome work! As for the version, not a problem at all, very common. From the nixpkgs manual:
> If a package is not a release but a commit from a repository, then the version part of the name must be the date of that (fetched) commit. The date must be in "YYYY-MM-DD" format. Also append "unstable" to the name - e.g., "pkgname-unstable-2014-09-23".

rexim commented:
@aanderse thank you so much for the quick response! Updated the version accordingly.

aanderse commented:
@GrahamcOfBorg build chatterino2

aanderse commented:
@rexim made 2 tiny changes before merging: consistent spacing in `longDescription` and removed redundant `stdenv.lib` text.

Thanks for your contribution to nixpkgs! :tada:
